---
title: "Data input from website"
author: "Jeremy Miller"
date: "March 8, 2019"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Overview

This script reads converts data downloaded from the Allen Institute Cell Types Database (http://celltypes.brain-map.org/rnaseq) into a format compatible for these scripts.

### Prior to running any code, download the data:
  
1. If needed, set the working directory first: e.g., `setwd("C:/Users/yourname/Desktop/analysis/")`  
2. Create a subfolder in your current working directory called `website_data`.  **This should be the same folder where you download everything in the `data` directory from `MTG_celltypes`.**
3. Download and unzip the following files in `data`.  
    a. Human MTG:  http://celltypes.brain-map.org/api/v2/well_known_file_download/694416044  
    b. Mouse VISp: http://celltypes.brain-map.org/api/v2/well_known_file_download/694413985  
    c. Mouse ALM:  http://celltypes.brain-map.org/api/v2/well_known_file_download/694413179 
_NOTE: These links are accurate as of April 2019.  If these data are moved in later releases, this code document will be updated accordingly - please post an Issue if needed.  Similarly, note that the dates below may need to be updated._  


```{r setwd}
homeFolder <- "../"  # UPDATE WITH CORRECT PATH
knitr::opts_knit$set(root.dir = homeFolder)
#setwd(homeFolder) # UPDATE WITH CORRECT PATH
```


### Load the relevant libraries

```{r load libraries, warnings=FALSE}
source(paste0(homeFolder,"R/Support_code0_functions.R"))
suppressPackageStartupMessages({
  library(feather)
  library(matrixStats)
  library(dplyr)
  library(edgeR)
  library(data.table)
})
options(stringsAsFactors=FALSE)
```
  
  
## Process the human MTG data
  
First we need to read the data into R. 
  
```{r read in MTG data}
tmp      <- fread(paste0(homeFolder,"website_data/human_MTG_2018-06-14_exon-matrix.csv"),header = T, sep = ',',verbose=FALSE)
exons    <- as.matrix(tmp[,2:dim(tmp)[2]])
rownames(exons) <- as.character(as.matrix(tmp[,1]))
tmp      <- fread(paste0(homeFolder,"website_data/human_MTG_2018-06-14_intron-matrix.csv"),header = T, sep = ',',verbose=FALSE)
introns  <- as.matrix(tmp[,2:dim(tmp)[2]])
rownames(introns) <- as.character(as.matrix(tmp[,1]))
geneInfo <- read.csv(paste0(homeFolder,"website_data/human_MTG_2018-06-14_genes-rows.csv"),row.names=1)
sampInfo <- read.csv(paste0(homeFolder,"website_data/human_MTG_2018-06-14_samples-columns.csv"),row.names=1)
```
  
Second, convert the meta-data files into formats consistent with the rest of the analysis.  Note that the MTG cluster colors (and other info) which is stored as a data file in `VENcelltypes`.  
  
```{r format MTG metadata}
# Omit cells with no class
kp <- sampInfo$cluster!="no class"

# Format the cluster info
anno <- auto_annotate(sampInfo[kp,])
anno$sample_id <- anno$sample_name

# Update the correct cluster colors and ids
load(paste0(homeFolder,"data/clusterInfoMTG.RData"))        #### May need to adjust location
anno$cluster_color <- clusterInfoMTG$cluster_color[match(anno$cluster_label,clusterInfoMTG$cluster_label)]
anno$cluster_id    <- clusterInfoMTG$cluster_id[match(anno$cluster_label,clusterInfoMTG$cluster_label)]
```
  
  
Next, convert the data into CPM(exons+introns) and format appropriately.    
  
```{r format MTG data}
## Calculate CPM
CPM <- cpm(introns[, kp] + exons[, kp])
rownames(CPM) <- rownames(geneInfo)
colnames(CPM) <- anno$sample_id

## Format appropriately
expr.data <- transpose(as.data.frame(CPM))
colnames(expr.data) <- rownames(CPM)
expr.data$sample_id <- anno$sample_id
```
  
Finally, output the results to feather files in the `MTG` directory.  
  
```{r output MTG data}
# Create MTG directory
dir.create(paste0(homeFolder,"MTG"))

# Write annotation file
write_feather(anno,paste0(homeFolder,"MTG/anno.feather"))

# Write data file
write_feather(expr.data,paste0(homeFolder,"MTG/data.feather"))
```
